code/code/spec/spec_objs.cc

```diff
@@ -6197,21 +6262,133 @@ int stickerBush(TBeing* ch, cmdTypeT cmd, const char*, TObj* o, TObj*) {
 }
 
 int rechargingWand(TBeing* ch, cmdTypeT cmd, const char*, TObj* o, TObj*) {
-  TWand* tw;
+  if (cmd != CMD_GENERIC_PULSE || !o || ::number(0, 49))
+    return false;
 
-  if (cmd != CMD_GENERIC_PULSE)
-    return FALSE;
+  ch = dynamic_cast<TBeing*>(o->equippedBy);
 
-  if (!(tw = dynamic_cast<TWand*>(o)))
-    return FALSE;
+  // The proc should only work when a mage mob or PC has the object equipped
+  if (!ch || !ch->hasClass(CLASS_MAGE))
```

Comment:
There's wands that have prayers and shaman rituals attached they aren't just mage tools. Seems like the proc is meant to convey that the item is magic not that the player is doing anything to recharge the item. This just seems wrong to me. How strongly do you feel about this?